mesonbuild/compilers/compilers.py

```diff
@@ -274,6 +274,16 @@ class CompileCheckMode(enum.Enum):
 clike_debug_args = {False: [],
                     True: ['-g']}  # type: T.Dict[bool, T.List[str]]
 
+class SanitizeOption(coredata.UserArrayOption):
+    def verify_value(self, value: T.List[str]) -> None:
+        super().verify_value(value)
+        if 'none' in value and len(value) > 1:
+            raise MesonException('b_sanitize=none cannot be combined with other values')
+        if int('thread' in value) + int('memory' in value) + int('address' in value) > 1:
```

Comment:
I think you could do something like `if len({'thread', 'memory', 'address'}.intersection(value)) > 1:`